client/go/outline/fetch.go

```diff
@@ -21,22 +21,22 @@ import (
 	"github.com/Jigsaw-Code/outline-apps/client/go/outline/platerrors"
 )
 
-// FetchResourceResult represents the result of fetching a resource located at a URL.
+// fetchResourceResult represents the result of fetching a resource located at a URL.
 //
 // We use a struct instead of a tuple to preserve a strongly typed error that gobind recognizes.
-type FetchResourceResult struct {
+type fetchResourceResult struct {
 	Content string
 	Error   *platerrors.PlatformError
 }
 
-// FetchResource fetches a resource from the given URL.
+// fetchResource fetches a resource from the given URL.
 //
 // The function makes an HTTP GET request to the specified URL and returns the response body as a
 // string. If the request fails or the server returns a non-2xx status code, an error is returned.
-func FetchResource(url string) *FetchResourceResult {
+func fetchResource(url string) *fetchResourceResult {
```

Comment:
*nit* Since we are not exporting this function directly, we can use the traditional Go style return type:

```suggestion
func fetchResource(url string) (string, error) {
```

And the caller can simply:

```
content, err := fetchResource(url)
return &InvokeMethodResult{
	Value: content,
	Error: platerrors.ToPlatformError(err),
}
```

Reply:
Done